### PR TITLE
New rawbuffer

### PR DESCRIPTION
**Pull request by savuor**

### CODE REVIEW

File: modules/vmfcore/include/vmf/types.hpp [hunk 1]
```diff
@@ -103,86 +104,45 @@ namespace vmf
...
-            }
-            else
-                data = nullptr;
-        }
+        explicit vmf_rawbuffer(const vmf_rawbuffer& other) : std::vector<char>(other)
+        { }
 
         vmf_rawbuffer(vmf_rawbuffer&& other)
         {
             *this = std::move(other);
```
Comment (apavlenko) [inline modules/vmfcore/include/vmf/types.hpp]: I suggest `std::swap()` here instead

File: modules/vmfcore/include/vmf/types.hpp [hunk 2]
```diff
@@ -103,86 +104,45 @@ namespace vmf
...
         }
 
-        ~vmf_rawbuffer()
+        vmf_rawbuffer& operator=(const vmf_rawbuffer& other)
```
Comment (apavlenko) [inline modules/vmfcore/include/vmf/types.hpp]: aren't you OK with the default compiler-generated code?

BTW, do you miss move/rvalue version?

File: modules/vmfcore/include/vmf/types.hpp [hunk 3]
```diff
@@ -103,86 +104,45 @@ namespace vmf
...
-        {
-            if(other.size > 0)
-            {
-                data = std::unique_ptr<char[]>(new char[other.size]);
-                memcpy(data.get(), other.data.get(), other.size);
-            }
-            else
-                data = nullptr;
-        }
+        explicit vmf_rawbuffer(const vmf_rawbuffer& other) : std::vector<char>(other)
```
Comment (apavlenko) [inline modules/vmfcore/include/vmf/types.hpp]: why `explicit`?

File: modules/vmfcore/include/vmf/types.hpp (hunk 1 above)
Comment (savuor) [inline modules/vmfcore/include/vmf/types.hpp]: fixed

File: modules/vmfcore/include/vmf/types.hpp (hunk 2 above)
Comment (savuor) [inline modules/vmfcore/include/vmf/types.hpp]: The operator=() is used in the next constructor and in std::swap() function.
Move version of operator=() created.

File: modules/vmfcore/include/vmf/types.hpp (hunk 3 above)
Comment (savuor) [inline modules/vmfcore/include/vmf/types.hpp]: You're right, `explicit` isn't needed. Fixed.

**Comment by apavlenko** [comment]
actually we have just a single test for `rawbuffer`,
looks like we should check all the methods...

**Comment by savuor** [comment]
@apavlenko tests were written for all constructors of `vmf_rawbuffer`

**Comment by apavlenko** [comment]
:+1:

**Pull request merged**